ipywebrtc/webrtc.py

```diff
@@ -71,93 +71,168 @@ def _valid_fps(self, proposal):
+        """
+        return cls(image=Image.from_file(filename), **kwargs)
+
+    @classmethod
+    def from_url(cls, url, **kwargs):
+        """Create a `ImageStream` from a url.
+        This will create a `ImageStream` from an Image using its url
+
+        Parameters
+        ----------
+        url: str
+            The url of the file that will be used for the .image trait.
+        **kwargs:
+            Extra keyword arguments for `ImageStream`
+        Returns an `ImageStream`.
+        """
+        return cls(image=Image.from_url(url), **kwargs)
 
 
 @register
 class VideoStream(MediaStream):
-    """Represents a media source by a video.
-
-    The `value` of this widget accepts a byte string. The byte string is the
-    raw video data that you want the browser to display.  You can explicitly
-    define the format of the byte string using the `format` trait (which
-    defaults to 'mp4').
-    If you pass `"url"` to the `"format"` trait, `value` will be interpreted
-    as a URL as bytes encoded in ascii.
-    """
+    """Represent a stream of a video element"""
     _model_name = Unicode('VideoStreamModel').tag(sync=True)
 
-    format = Unicode('mp4', help="The format of the video.").tag(sync=True)
-    value = Bytes(None, allow_none=True, help="The video data as a byte string.").tag(sync=True)
-    play = Bool(True, help='Plays the video or pauses it.').tag(sync=True)
-    loop = Bool(True, help='When true, the video will start from the beginning after finishing.').tag(sync=True)
+    video = Instance(
+        Video,
+        allow_none=False,
+        help="An ipywidgets.Video instance that will be the source of the media stream."
+    ).tag(sync=True, **widget_serialization)
+    play = Bool(True, help='Plays the videostream or pauses it.').tag(sync=True)
 
     @classmethod
-    def from_file(cls, f, **kwargs):
-        """Create a `VideoStream` from a local file or file object.
+    def from_file(cls, filename, **kwargs):
+        """Create a `VideoStream` from a local file.
 
         Parameters
         ----------
-        f: str or file
-            The path or file object that will be read and its bytes assigned
-            to the value trait.
+        filename: str
+            The location of a file to read into the value from disk.
         **kwargs:
             Extra keyword arguments for `VideoStream`
-        Returns an `VideoStream` with the value set from the content of a file.
+        Returns an `VideoStream`.
         """
-        if isinstance(f, string_types):
-            with open(f, 'rb') as f:
-                return cls(value=f.read(), **kwargs)
-        else:
-            if 'format' not in kwargs:
-                ext = os.path.splitext(f)[1]
-                if ext:
-                    kwargs['format'] = ext[1:]  # remove the .
-            return cls(value=f.read(), **kwargs)
+        autoplay = True
+        if kwargs.get('play') is not None:
+            autoplay = kwargs.get('play')
```

Comment:
`autoplay = kwargs.get('play', True)` is shorter. However, why would it autoplay? Maybe play it once captureStream is called? That would be more in line with CameraStream.

Reply:
I'll try that. But if I play the video once `captureStream` is called, it will play it every time I create a new view of the VideoStream and I think it will start the video from the beginning again.

Reply:
captureStream can return a promise right, why not create the view in the captureStream, and play it there, than it only happens once.

Reply:
I disagree, if we create a new view of the `Video` every time I call `captureStream`, then I will lose synchronicity between `VideoStream` views, because they don't rely on the same `Video` view. This means that the `play` trait won't make any sense, and if I record a `VideoStream` with `VideoRecorder` I won't know which view is recorded.

Speaking about synchronicity, we discussed with Sylvain this morning about Video widget, and we agreed on the fact that the current implementation of  `VideoStream` should be the implementation of the `Video` widget. We currently don't have synchronicity between views for the `Video` widget, but if we have only one `video` element as part of the model, and the views are only streams of this `video` element, then we have synchronicity. And then it makes sense to have a `play` and `progress` traits in `Video` widget, which doesn't make sense right now.

Reply:
No, I mean, the captureStream should return this single promise, but it's easier to create the view in the chain of that promise.

I don't agree with the Video having synchronicity, if you create a video tags with controls, you can change the slider to where it should play, you lose that when you show a MediaStream instead.

Reply:
Ok so I agree with what you are saying, but only if we create this promise once

Reply:
Indeed, we create the promise once, and in it we create the view. (I'd say, write some js unittests for it).